src/vorta/views/archive_tab.py

```diff
@@ -216,42 +221,55 @@ def check_action(self):
                 archive_name = archive_cell.text()
                 params['cmd'][-1] += f'::{archive_name}'
 
-        thread = BorgCheckThread(params['cmd'], params, parent=self.app)
+        thread = BorgCheckThread(params['cmd'], params)
         thread.updated.connect(self._set_status)
         thread.result.connect(self.check_result)
         self._toggle_all_buttons(False)
         thread.start()
+        return thread
 
     def check_result(self, result):
         if result['returncode'] == 0:
             self._toggle_all_buttons(True)
 
+    def job_prune_action(self):
+        profile = self.profile()
+        QApplication.instance().scheduler.jobs_manager.add_job(
+            FuncJob(self.prune_action, site=profile.repo.id))
+
     def prune_action(self):
         params = BorgPruneThread.prepare(self.profile())
         if params['ok']:
-            thread = BorgPruneThread(params['cmd'], params, parent=self.app)
+            thread = BorgPruneThread(params['cmd'], params)
             thread.updated.connect(self._set_status)
             thread.result.connect(self.prune_result)
             self._toggle_all_buttons(False)
             thread.start()
+            return thread
         else:
             self._set_status(params['message'])
 
     def prune_result(self, result):
         if result['returncode'] == 0:
             self._set_status(self.tr('Pruning finished.'))
-            self.list_action()
+            self.job_list_action()
         else:
             self._toggle_all_buttons(True)
 
+    def job_list_action(self):
+        profile = self.profile()
+        QApplication.instance().scheduler.jobs_manager.add_job(
+            FuncJob(self.list_action, site=profile.repo.id))
```

Comment:
So this is the new way to run a Borg command? I'm surprised this still works. Isn't the function running in another context or is it the same in another Thread? Where did you learn or see this pattern? Looks really useful for GUI apps.

Reply:
What we really want is async/await with QThreads, right? This lib here is kinda similar: https://github.com/CabbageDevelopment/qasync/blob/master/qasync/__init__.py#L166

Reply:
I'm not sure to understand. 
It's the same function run in another thread. FuncJob is a convenient way to run a func in  a job. Usually, you have to inherits Job and override run method. Job is defined in scheduler.py 
Its inspired from the command pattern which is indeed useful for gui.

Reply:
True. Async/await wouldn't have a queue. It just runs the function without blocking the main thread (or UI).